runtime/lua/vim/lsp/tagfunc.lua

```diff
@@ -23,36 +30,42 @@ local function query_definition(pattern)
   local add = function(range, uri, offset_encoding)
     table.insert(results, mk_tag_item(pattern, range, uri, offset_encoding))
   end
-  for client_id, lsp_results in pairs(results_by_client) do
+  for client_id, lsp_results in pairs(results_by_client or {}) do
     local client = lsp.get_client_by_id(client_id)
+    local offset_encoding = client and client.offset_encoding or 'utf-16'
     local result = lsp_results.result or {}
     if result.range then -- Location
       add(result.range, result.uri)
-    else -- Location[] or LocationLink[]
+    else
+      result = result --[[@as (lsp.Location[]|lsp.LocationLink[])]]
       for _, item in pairs(result) do
         if item.range then -- Location
-          add(item.range, item.uri, client.offset_encoding)
+          add(item.range, item.uri, offset_encoding)
         else -- LocationLink
-          add(item.targetSelectionRange, item.targetUri, client.offset_encoding)
+          add(item.targetSelectionRange, item.targetUri, offset_encoding)
         end
       end
     end
   end
   return results
 end
 
+---@param pattern string
+---@return table[]
 local function query_workspace_symbols(pattern)
   local results_by_client, err =
     lsp.buf_request_sync(0, ms.workspace_symbol, { query = pattern }, 1000)
   if err then
     return {}
   end
   local results = {}
-  for client_id, symbols in pairs(results_by_client) do
+  for client_id, responses in pairs(results_by_client or {}) do
```

Comment:
Isn't results_by_client guaranteed to be non-nil. If so then it just needs an `aasert`.